### DISCUSSION TITLE
Why are you making Homebrew into MacPorts?

### DISCUSSION THREAD

**Discussion by bcardarella** [Homebrew/discussions#2801] [Everyday usage]
Upgrading ffmpeg installed/upgraded the following dependencies:

> openexr, webp, jpeg-xl, libvmaf, aom, freetype, libunistring, p11-kit, gdbm, sqlite, python@3.9, glib, libx11, gobject-introspection, harfbuzz, cjson, cmocka, mbedtls, librist, rav1e, rubberband, sdl2, zstd, libarchive and tesseract

This is ridiculous and the entire reason why people abandoned MacPorts in favor of Homebrew in the first place. Upgrading a package shouldn't install 40 other packages. Stop this. Don't become MacPorts.

This is not a troll, please don't dismiss it as such. This is a serious complaint and criticism of how complex and bloated Homebrew has become. The dev community wants simple and lean package managers. Not this obese monstrosity that Homebrew has become.

**Comment by tjluoma** [comment]
_(I’m not a developer/maintainer, just a user of Homebrew… and MacPorts, on different computers.)_

If those are **dependencies**…then they need to be installed/updated when the main package is installed/updated, or else it won’t work. What option do the maintainers have besides including them?

`ffmpeg` is a large and complicated program with a lot of features. It’s not as if you installed something simple and Homebrew decided to install 10 unrelated items along with it.

MacPorts would have to do the same thing. as would any package manager, so, again, I’m not sure what you’re suggesting the alternative is.

You say that your message is not a troll, but the title is written in an aggressive, almost click-bait manner, you refer to the work of the Homebrew maintainers as an “bloated…obese monstrosity”, and you are claiming to speak for the entire developer community, both expressing what you think they want and why they left MacPorts. No support is given to these claims, and you’ve offered no actionable suggestions. None of these things are going to help your cause to be taken seriously.

Imagine walking up to a developer in person and telling them that their work is a ridiculous, bloated, obese monstrosity. Do you really think that they would be inclined to listen to anything that you had to say, or would it be far more likely that you would be considered, at best, arrogant, rude, and entitled?

Now imagine that at the end of that I added: “This is a serious complain and criticism of your post, so please don’t dismiss it as anything else, but take it to heart and make the necessary changes that I deem necessary.” How inclined are _**you**_ to do so?

**Reply by bcardarella** [reply]
The entire marketing around Homebrew was to pit it against MacPorts when it came out and Homebrew *itself* leaned heavily on the fact that people were sick of having to install every dependency under the sun when installing packages via MacPorts. Now, Homebrew does the exact same thing. So no, it isn't trolling and it isn't arrogant, rude, or entitled. I'm asking the project to return to its intended roots. Trying to gaslight this request the way you have is in itself arrogant, rude, and entitled.

**Reply by tjluoma** [reply]
Your understanding of the word “gaslight” seems even worse than your understanding of the word “dependency”.

All I did was restate your own words back to you, and you considered them arrogant, rude, and entitled.

**Reply by bcardarella** [reply]
Your ability to focus on the hyperbolic while ignoring the more salient issue at hand is blowing my mind. Best of luck to you.

**Reply by carlocab** [reply]
Hyperbole does tend to be a lot more salient than... not hyperbole, so I'm really not sure what you were expecting.

**Reply by bcardarella** [reply]
I'm hoping that this nonsense of continuing to add bloat to homebrew will stop. Clearly I'm not in the majority in this.

**Reply by carlocab** [reply]
How do you even know that this is all just bloat, as opposed to dependencies that enable useful features in `ffmpeg` that other users use?

**Reply by bcardarella** [reply]
@carlocab forgive me if I'm not entirely trusting that the homebrew core team doesn't have the best historical record of taking community feedback into consideration and ending up on the right side of history. See an issue I opened back in 2016 and how well that has aged in light of the current state and abuse of data collection on the Internet: https://github.com/Homebrew/brew/issues/142

So If I'm coming in hot on this issue, it's because I've tried the other way in the past.

**Reply by carlocab** [reply]
Even if you were right about that issue (and I'm not taking a position on whether you were or weren't), I don't think it justifies the rudeness of your earlier messages.

I wasn't even with the project back then, so I'm not at all responsible for that decision.

That also doesn't really answer my earlier question:

> How do you even know that this is all just bloat, as opposed to dependencies that enable useful features in `ffmpeg` that other users use?

**Reply by bcardarella** [reply]
I think it is just systemic in the DNA of this project that the maintainers are defensive about any slight hint of criticism. The current installations are too big too much and IMO not in the spirit of what spawned the project in the first place. Defending this as being the best installation path without any room to discuss the merits of the dependency bloat tells me everything I need to know. I'll move onto another package manager that doesn't have the weight that homebrew does.

**Reply by tjluoma** [reply]
“Any slight hint of criticism” does not come anywhere near close to what you did with this thread.

You came in arrogant and rude, and then blamed Homebrew for being defensive.

**Reply by bcardarella** [reply]
Do I really need to go through the issue tracker and come up with the overwhelming number of examples? Get off your horse

**Reply by tjluoma** [reply]
Still waiting to hear a concrete recommendation as to what dependencies of `ffmpeg` you can declare with absolute certainty are not needed.

You have not shown any actual evidence that there _is_ bloat other than your say so.

**Reply by tjluoma** [reply]
"Your ability to focus on the hyperbolic while ignoring the more salient issue at hand is blowing my mind."

Your message was mostly hyperbolic, and when I've asked for specifics about the salient issues, you have ignored them entirely.

Good luck convincing anyone to listen to you.

**Reply by carlocab** [reply]
I'm actually trying to discuss it, but you're refusing to engage in the discussion. See also my comment here https://github.com/Homebrew/discussions/discussions/2801#discussioncomment-1985434 where I do discuss the merits of the added dependencies vs cutting them.

I'm also still interested in hearing which of the dependencies really are bloat that can be cut without breaking someone's usage of a formula.

Not sure where that says there's no room to discuss these dependencies, but ok.

**Reply by tjluoma** [reply]
"Get off your horse" says man who rode in on a giant horse.

No, you don't need to go through the issue tracker. I want you to tell me which dependencies of `ffmpeg` are bloat and why.

Stop expecting people to treat your opinion as gospel truth.

**Comment by SMillerDev** [comment]
I expect the main reason is that when homebrew started, macOS came with quite some software. 10+ years later it comes with pretty much the same software versions as back then but the rest of the software world has moved on. If you want less dependencies, tell the projects you're installing to drop features and work with ancient versions of dependencies. Then we can go back to using the macOS system parts.

**Reply by bcardarella** [reply]
Why is it that, considering we're all on either Intel or M1/2 chips, precompiled binaries are not favored? We're dealing with a small set of chips that would make the installation experience far better and less tedious than what Homebrew has become.

**Reply by SMillerDev** [reply]
Precompiled binaries are the only supported way of installing homebrew software, your suggestion is exactly what homebrew has become.

**Comment by SMillerDev** [comment]
> This is not a troll, please don't dismiss it as such. This is a serious complaint and criticism of how complex and bloated Homebrew has become. The dev community wants simple and lean package managers. Not this obese monstrosity that Homebrew has become.

I'd like to point you to the code of conduct and request that you follow it because being a dick to volunteers is not how you motivate people to help you.

**Comment by carlocab** [comment]
There are trade-offs here that we need to make:
1. Formulae should be packages so that they provide most of the features most users want to use. This tends to add to the number of dependencies that a formula has.
2. We also want dependencies to be minimal. The primary reasons for this used to be that minimal dependency trees meant a) less time spent compiling things (and having to diagnose the eventual failures associated with them), and b) less space used up in having to keep them installed.

We install everything from bottles now (unless you're running an unsupported configuration -- we can't help you with that), so reason (a) above is a non-issue for users. Disk space today is also quite cheap, which makes reason (b) much less compelling than it used to be. Note that the formulae you list are ~350 MB on disk, so, while not nothing, don't seem a huge deal either.

Note also, that you seem to have misunderstood what distinguishes Homebrew from MacPorts: it's not that we design our packages to be light. It's that we don't duplicate system-provided dependencies except when we need to. The only thing provided by your system that Homebrew has installed is `sqlite` -- and that's because we ship a newer version. (No, your system does not provide `python3`. That's only installed if you install the CLT, and that version is not suitable.)

**Comment by iMichka** [comment]
Locking this discussion. This is going nowhere and I don't like the tone used here.
Maintainer time is better used than answering to this thread.

The different topics mentioned here where already discussed endlessly, in this discussion, in other issues and other places.

Let me also remember you our license terms:

```
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED.
```

We, as homebrew maintainers, think that what we are doing is the right thing, and won't change brew's behaviour regarding your comment. Sorry, passing on this.
